images/hub/jupyterhub_config.py

```diff
@@ -50,42 +76,105 @@
 
```

Comment:
Line 75 will no longer function because this was removed from the configmap:

```
  {{- /* KubeSpawner */}}
  kubespawner.common-labels: |
    {{- $_ := merge (dict "heritageLabel" "jupyterhub") . }}
    {{- include "jupyterhub.commonLabels" $_ | nindent 4 }}
```

Reply:
I'm trying to figure out the right thing to do here, and I see two options:

1. implement the correct labels in Python (needs Chart/Release inputs, but we need to subset them, since they contain timestamps/hash versions, which shouldn't be included in the configmap lest they trigger relaunch of the Hub)
2. preserve this common-labels passthrough as-is

My inclination is to reimplement in Python, with filtering out things like Chart.version and Release.timestamp to avoid unnecessary churn. What do you think?

Reply:
@minrk +1 to reimplement in Python and scrap the helper. This is one of these helpers that is over the top. It is defined, and used twice. See below.

**Definition:**
https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/ef4b0f757ac3314e2679b16a66373a372df960ae/jupyterhub/templates/_helpers.tpl#L130-L142

**Usage 1:**
https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/ef4b0f757ac3314e2679b16a66373a372df960ae/jupyterhub/templates/_helpers.tpl#L145-L152

**Usage 2:**
https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/ef4b0f757ac3314e2679b16a66373a372df960ae/jupyterhub/templates/hub/configmap.yaml#L259-L262

Reply:
This is fixed now, I believe. I do think we should revamp the helpers (#964), but save that for a later single-purpose PR.